### PR TITLE
wip

### PR DESCRIPTION
**Pull request by pedroliveirav**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx [hunk 1]
```diff
@@ -310,11 +282,70 @@ const SetupAnalyticsCloudPage = ({
...
+				);
+			}
+
+			handleDataSubmit();
 			setIsLoadingSubmitButton(false);
 
 			handlePage(true);
-		} catch {
-			setIsLoadingSubmitButton(false);
+		} catch (error) {
```
Comment (ryanschuhler) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:323]: @pedroliveirav if we are properly comparing old and new contact arrays wouldnt we exclude the possibility of hitting the conflict code? or what circumstance would allow us to see this error in the ui?

Separate note, we should either way write playwrite tests to test the conflict edge case and ensure we either dont get the error or it shows up when expected

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupDXPCloudForm/index.jsx [hunk 2]
```diff
@@ -187,6 +190,100 @@ const SetupDXPCloudPage = ({
...
+				},
+			});
+
+			if (featureFlags.includes('LPS-187767')) {
+				const notificationTemplateService = new NotificationQueueService(
+					client
+				);
+
+				try {
+					await notificationTemplateService.send(
```
Comment (ryanschuhler) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupDXPCloudForm/index.jsx:263]: I am not completely sure everything that is happening here, but feels odd to be sending notifications/emails from the front end. What is the use case for getting into this try block? and is it something we could/should do with workflow/object actions?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx (hunk 1 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:323]: All the contacts handled are on the portal side, but the conflict occurs when the user assigned to a specific role already has that role in the provisioning, so when the error is a conflict, the requests made are only on the liferay side. I don't know if my answer satisfies all your doubts.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupDXPCloudForm/index.jsx (hunk 2 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupDXPCloudForm/index.jsx:263]: It's an e-mail notification structured via the notification template if I'm not mistaken, but this structure already existed, I just put the whole process inside a function so that I could call the function too when I'm in the error condition due to conflict, so I kept this structure. 

We're already using object actions in this notification template too.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx [hunk 3]
```diff
@@ -310,11 +282,70 @@ const SetupAnalyticsCloudPage = ({
...
+			if (error.cause === STATUS_CODE.conflict) {
+				try {
+					await actLiferayContact(
```
Comment (kylembischof) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:326]: why are we trying `actLiferayContact` again after the conflict error? can we reorder it in a way where we can improve these if statements and try catches?

Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:326]: wasn't it decided that we would handle the request when the error was due to a conflict? That's why, when we're in this condition, we try actLiferayContact. The provisioning side already has the role, so we need to accept the role on the Liferay side.

Comment (kylembischof) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:326]: yeah you have the right idea, but is there a way to clean it up visually so we don't have to nest try/catches, call functions multiple times, set `setIsLoadingSubmitButton(false)` so many times, things like that. 
while something may be conceptually correct, it could still be optimized visually or implementation wise

Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:326]: Okay, I thought I had misunderstood, I'll try to find a better way to run this process.

**Comment by jpince** [comment]
@pedroliveirav is this Pull Request still relevant? or can it be closed?

**Comment by pedroliveirav** [comment]
> @pedroliveirav is this Pull Request still relevant? or can it be closed?

Given the discussion in the PR review, I'm going to need to refactor the code relating to this PR, so I'm going to close this PR and resend it when it's completed.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx (hunk 3 above)
Comment (jennyhchen) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:326]: @ryanschuhler Because there are setStates within all the try/catches, is it possible to abstract these out? From our backend perspective, we do want to limit duplicate code, and in order to do that, we usually pass through variables in methods. However, I'm unfamiliar as to the best practices for frontend. Please let me know if I'm not understanding this correctly, but I remember react training mentioned we shouldn't be passing around states unless absolutely necessary and they really should stay within their own component/files.

Is there any way to reduce code duplication on frontend code within this example on Pedro's PR? I did read up on ErrorBoundaries but even those don't really touch useStates if you're expecting certain frontend behavior changes.
